layout/partials/ctm.html

```diff
@@ -1,14 +1,39 @@
 {{#if @root.flags.javascript}}
-var classes = document.documentElement.className;
-classes = classes.replace(/\bno-js\b/, 'js');
-
-// HACK: Make FB in-app browser on iOS not cut the mustard, temporarily
-var isFBInAppBrowser = /OS [1-9].* like Mac OS X/.test(navigator.userAgent) && /(FBAN|FBAV)/.test(navigator.userAgent);
-window.cutsTheMustard = (typeof Function.prototype.bind !== 'undefined') && !isFBInAppBrowser;
-if (window.cutsTheMustard) {
-	classes = classes.replace(/\bcore\b/, 'enhanced');
-}
-document.documentElement.className = classes;
+(function () {
+	var classes = document.documentElement.className;
+	classes = classes.replace(/\bno-js\b/, 'js');
+
+	// HACK: make the following browsers core as they're unsupported by the polyfill service
+	var unsupported = {
+		'iOS Facebook in-app browser': function (ua) {
+			var iOS = /OS [1-9].* like Mac OS X/.test(ua);
+			var fb = /(FBAN|FBAV)/.test(ua);
+			return iOS && fb;
+		},
+		'Samsung frankenchromium': function (ua) {
+			var android = /Android/.test(ua);
+			var samsung = /SamsungBrowser/.test(ua);
+			return android && samsung;
+		}
+	};
+
+	var isUnsupported = false;
```

Comment:
I'd prefer it to start with `isSupported = true;` and then set to false in the loop. Easier to read